### DISCUSSION TITLE
Energy Based Damage for Ductile Fracture in Peridynamics

### DISCUSSION THREAD

**Discussion by ppandit95** [idaholab/moose#19388] [Q&A Modules: General] [answered]
Dear MOOSE Community

I wish to simulate the tensile test for a ductile material and get the desired stress strain curve but I was wondering if energy based bond damage model is implemented as in the plastic region , a critical stretch based model wont work due to non linearities.Any headers in this regard will be helpful.

Moreover,I donot know whether a isotropic hardening model is available or not.

Many Thanks
Pushkar

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this for peridynamics?
There are isotropic hardening models, for example
https://mooseframework.inl.gov/source/materials/IsotropicPowerLawHardeningStressUpdate.html

I d look in the tensor mechanics documentation for what you want:
https://mooseframework.inl.gov/modules/tensor_mechanics/index.html#!

Guillaume

**Reply by hugary1995** [reply]
Hi Pushkar,

The Peridynamics module is currently limited to elastic constitutive models (a few simple damage models are a notable exception). As far as I can tell, the Peridynamics module doesn't have any plasticity models yet. However, implementing new material models should mostly be copy-pasting from the tensor mechanics module. So you can follow the links @GiudGiud posted above.

**Reply by ppandit95** [reply]
thanks @GiudGiud and @hugary1995 for directing to the relevant sources which can be used for modeling plasticity with isotropic hardening .

**Reply by jiangwen84** [reply]
I remembered @hchen139 implemented creep and/or plasticity for the NOSPD formulation. @hchen139 Could you comment on this?

**Reply by hchen139** [reply]
NOSPD can incorporate any available inelastic material model from tensor mechanics module, such as creep and plasticity models. As for damage models for PD, the only one available is bond stretch-based model. You will need to develop your own specific damage model based on current capability of the PD module in MOOSE.

**Reply by ppandit95** [reply]
thanks @jiangwen84 and @hchen139 for elaborating on the issue of damage modelling in ductile materials ....
